src/dsl/dsl.jl

```diff
@@ -5,7 +5,9 @@ const DSL_ARG_GRAD_ANNOTATION = :grad
 const DSL_RET_GRAD_ANNOTATION = :grad
 const DSL_TRACK_DIFFS_ANNOTATION = :diffs
 const DSL_NO_JULIA_CACHE_ANNOTATION = :nojuliacache
-const DSL_MACROS = Set([Symbol("@trace"), Symbol("@param")])
+
+# the set of macros we should not macroexpand before parsing
+const DSL_MACROS = Set([Symbol("@trace"), Symbol("@param"), Symbol("@gen")])
```

Comment:
I think we can omit `@gen` from this list, because the manual macro-expansion is only going to be applied to the function body, not the whole function.